user/service.py

```diff
@@ -6,12 +6,17 @@
 import user.dao as user_dao
 import user_token.service as token_service
 from db_python_util.db_classes import User
+from db_python_util.db_exceptions import EntityNotFoundException
 
+import be_exceptions.validation_exceptions as validation_exceptions
 def get(user_id: str = None) -> List[User] | Optional[User]:
     """Get a user by ID, or all users if no ID is given."""
 
     if user_id is not None:
-        return user_dao.get_by_id(user_id)
+        try:
+            return user_dao.get_by_id(user_id)
+        except EntityNotFoundException as e:
+            raise validation_exceptions.MissingException(f"The given user ID '{user_id} does not correspond to an existing user.")
```

Comment:
There's really no harm in this except for maybe the extra lines but I guess I'm just a little confused why the service layer needs to replace the data-layer exceptions with its own. Not a blocker by any means though.

```suggestion
            raise validation_exceptions.MissingException(f"The given user ID '{user_id}' does not correspond to an existing user.")
```

Reply:
I agree with doing this.  It allows for service layer logic to be applied if necessary and provides an abstraction over the data layer generally.